### PR TITLE
Fix crash on x86_64 emulators 

### LINKED ISSUE #17265: Can't install e2e apk on Android 12 and higher
# Bug Report

## Problem 
To keep our e2e tests up to date we need to use relevant Android version. SauceLabs emulators are x86_64 only. And now it's not possible to install and use e2e application on such emulators with Android 12 or more.

#### Expected behavior
The e2e apk is installed on Android 12

#### Actual behavior
Screenshot 2023-09-12 at 17 21 13

### Reproduction

1. Create an Android emulator with x86 or x86_64 ABI and API level 31 or more
2. Try installing e2e application 

### Additional Information

- Status version: nightly e2e
- Operating System: Android 12 and higher 

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by siddarthkay**
This PR
- points to status-go branch where crash is fixed for x86_64 emulators
- attempts to explicitly build also for x86_64 ABI 

fixes #17265

#### Platforms
- Android

##### Functional
- 1-1 chats
- public chats
- group chats
- wallet / transactions
- dapps / app browsing
- account recovery
- new account
- user profile updates
- networks
- mailservers
- fleet
- bootnodes

##### Non-functional
- battery performance
- CPU performance / speed of the app
- network consumption

##### Todos

- [x] https://github.com/status-im/status-mobile/issues/18086
- [x] https://github.com/status-im/status-mobile/issues/17678

status: ready

### CODE REVIEW

**Review by qfrank — APPROVED** [review]
finally!

**Comment by qfrank** [comment]
@siddarthkay tested on my local x86_64 emulator, i'm able to create account now, but It seems that this Android APK version has lost its network communication capability. e.g. make contact request not work anymore

**Comment by siddarthkay** [comment]
Thanks for testing locally @qfrank!
I will try sending a contact request and see what went wrong.

**Comment by siddarthkay** [comment]
Doesn't also hurt to run E2E :)

**Comment by yevh-berdnyk** [comment]
Issues from failed e2e tests:

- There is one crash after getting the app from background: https://app.eu-central-1.saucelabs.com/tests/241b28787abf43e18a27d62b3a58f8b6?auth=1cf4f7240a49d2a962e4f78d88b00142#63
Logs: [logcat](https://drive.google.com/file/d/19f10TQFz8-P41hQz6FPetNtFYAi88l8a/view?usp=sharing)     [geth.log](https://github.com/status-im/status-mobile/files/13219110/test_activity_center_contact_request_decline_geth2.log)

- Invited member is failed waiting to join an open community for 60 seconds: https://app.eu-central-1.saucelabs.com/tests/cedaffa18b3b4bcbb3599a9fcea74133?auth=f25ad58526407273f0c66d2f5e81b5ac#98 
Logs: [logcat](https://drive.google.com/file/d/1NJzngXvjSZjEN4D7cil_Mk00wpHlpJav/view?usp=sharing)   [geth.log](https://github.com/status-im/status-mobile/files/13219450/TestActivityMultipleDevicePRTwo_geth1.log)


- Community channel view was not loaded for an invited member: https://app.eu-central-1.saucelabs.com/tests/cbb67591c55941d997cd6135a6288cb7?auth=82eeeed683a281db31beeaa6e3defee8#147
Logs: [logcat](https://drive.google.com/file/d/1hug73J0i--uQ3yR-zUO395c7saUgVjYi/view?usp=sharing)  [geth.log](https://github.com/status-im/status-mobile/files/13219255/test_community_mentions_push_notification_geth1.log)

File: test/appium/tests/base_test_case.py
Comment (siddarthkay) [inline test/appium/tests/base_test_case.py]: @yevh-berdnyk : just curious, do we have `platformVersion` 14 available at appium/saucelabs ?

**Comment by siddarthkay** [comment]
Trying to replicate Crash 1 

> There is one crash after getting the app from background

File: test/appium/tests/base_test_case.py
Comment (yevh-berdnyk) [reply test/appium/tests/base_test_case.py]: Yes, I've just ran one of the tests on Android14 and it passed

File: android/gradle.properties
Comment (jakubgs) [inline android/gradle.properties:45]: Shouldn't we just drop `x86` then?

**Review by jakubgs — APPROVED** [review]
Looks fine, but make sure to get a PR for `switch_e2e_to_x86_64` merged and then tagged.

File: android/gradle.properties
Comment (siddarthkay) [reply android/gradle.properties:45]: Hmm, worth exploring, I'll test removing it locally

**Comment by siddarthkay** [comment]
https://github.com/status-im/status-go/pull/4233 is approved.
However for status-go related PRs we make a corresponding status-mobile PR ( which is this one )
And when the status-mobile PR is QA approved we then merge status-go PR. 
Then push tag.
Then merge status-mobile PR with the new tag.

File: android/gradle.properties
Comment (siddarthkay) [reply android/gradle.properties:45]: modifying this field had no effect actually, we'll have to update 
https://github.com/status-im/status-mobile/blob/develop/nix/mobile/android/release.nix#L24

**Comment by yevh-berdnyk** [comment]
> Issues from failed e2e tests:
> 
> * There is one crash after getting the app from background: https://app.eu-central-1.saucelabs.com/tests/241b28787abf43e18a27d62b3a58f8b6?auth=1cf4f7240a49d2a962e4f78d88b00142#63
>   Logs: [logcat](https://drive.google.com/file/d/19f10TQFz8-P41hQz6FPetNtFYAi88l8a/view?usp=sharing)     [geth.log](https://github.com/status-im/status-mobile/files/13219110/test_activity_center_contact_request_decline_geth2.log)
> * Invited member is failed waiting to join an open community for 60 seconds: https://app.eu-central-1.saucelabs.com/tests/cedaffa18b3b4bcbb3599a9fcea74133?auth=f25ad58526407273f0c66d2f5e81b5ac#98
>   Logs: [logcat](https://drive.google.com/file/d/1NJzngXvjSZjEN4D7cil_Mk00wpHlpJav/view?usp=sharing)   [geth.log](https://github.com/status-im/status-mobile/files/13219450/TestActivityMultipleDevicePRTwo_geth1.log)
> * Community channel view was not loaded for an invited member: https://app.eu-central-1.saucelabs.com/tests/cbb67591c55941d997cd6135a6288cb7?auth=82eeeed683a281db31beeaa6e3defee8#147
>   Logs: [logcat](https://drive.google.com/file/d/1hug73J0i--uQ3yR-zUO395c7saUgVjYi/view?usp=sharing)  [geth.log](https://github.com/status-im/status-mobile/files/13219255/test_community_mentions_push_notification_geth1.log)

Regarding the third issue from this comment - here is an open bug: https://github.com/status-im/status-mobile/issues/17801
But looks like here, in this PR, it happens much more often

**Comment by yevh-berdnyk** [comment]
> > Issues from failed e2e tests:
> > 
> > * There is one crash after getting the app from background: https://app.eu-central-1.saucelabs.com/tests/241b28787abf43e18a27d62b3a58f8b6?auth=1cf4f7240a49d2a962e4f78d88b00142#63
> >   Logs: [logcat](https://drive.google.com/file/d/19f10TQFz8-P41hQz6FPetNtFYAi88l8a/view?usp=sharing)     [geth.log](https://github.com/status-im/status-mobile/files/13219110/test_activity_center_contact_request_decline_geth2.log)
> > * Invited member is failed waiting to join an open community for 60 seconds: https://app.eu-central-1.saucelabs.com/tests/cedaffa18b3b4bcbb3599a9fcea74133?auth=f25ad58526407273f0c66d2f5e81b5ac#98
> >   Logs: [logcat](https://drive.google.com/file/d/1NJzngXvjSZjEN4D7cil_Mk00wpHlpJav/view?usp=sharing)   [geth.log](https://github.com/status-im/status-mobile/files/13219450/TestActivityMultipleDevicePRTwo_geth1.log)
> > * Community channel view was not loaded for an invited member: https://app.eu-central-1.saucelabs.com/tests/cbb67591c55941d997cd6135a6288cb7?auth=82eeeed683a281db31beeaa6e3defee8#147
> >   Logs: [logcat](https://drive.google.com/file/d/1hug73J0i--uQ3yR-zUO395c7saUgVjYi/view?usp=sharing)  [geth.log](https://github.com/status-im/status-mobile/files/13219255/test_community_mentions_push_notification_geth1.log)
> 
> Regarding the third issue from this comment - here is an open bug: #17801 But looks like here, in this PR, it happens much more often

More logs:
In this test there were only one user, he created a community and got this loading chat view when navigated to the community channel: https://app.eu-central-1.saucelabs.com/tests/df28294479324662bc976fb9d6fd14df?auth=e83c42663eb7fe2afc04e22ea3f30b83#105
[logcat](https://drive.google.com/file/d/1THO1EG5mErjdjU_ytATO59yPK_FEbZjA/view?usp=sharing)
[geth.log](https://github.com/status-im/status-mobile/files/13252029/test_community_copy_and_paste_message_in_chat_input_geth1.log)

File: android/gradle.properties
Comment (yakimant) [reply android/gradle.properties:45]: Probably even higher - Jenkinsfile/Makefile.

**Review by yakimant — APPROVED** [review]

**Comment by yevh-berdnyk** [comment]
Regarding e2e test failures a lot of tests failed because of an issue with images on Android12 emulators from SauceLabs. It’s already reported and I’m waiting a fix from them. Also couple of failures are already known and not related to this PR.
But there are 2 specific issues here.
### Issue 1. Can't restore user's data from waku backup
**Steps:**
1. Recover user using a recovery phrase: `staff actress trash route grab crime leaf uniform dizzy reform issue keep` 
2. Check communities tab and contacts list

**Actual result:**
No communities and no contacts are restored
**Expected result:**
4 communities and 2 contacts are restored
[logcat.log.zip](https://github.com/status-im/status-mobile/files/13571535/logcat.2.log.zip)
[geth.log.zip](https://github.com/status-im/status-mobile/files/13571683/test_restore_multiaccount_with_waku_backup_remove_switch_geth1.1.log.zip)
Note: I can't reproduce this issue on real devices (ios builds and Android universal apk) but it's not happening in e2e builds from other branches 

### Issue 2. Can't join an open community when node owner is offline
**Steps:**
1. User A creates an open community and invites User B
2. User A goes offline
3. User B requests to join this community
4. User A goes back online

**Actual result:**
Community is shown as pending endlessly 
**Expected result:**
User B has joined the community
[user_A_logcat.log.zip](https://github.com/status-im/status-mobile/files/13572071/user_A_logcat.log.zip)
[user_B_logcat.log.zip](https://github.com/status-im/status-mobile/files/13572073/user_B_logcat.log.zip)
[user_A_geth.log.zip](https://github.com/status-im/status-mobile/files/13572090/user_A_geth.log.zip)
[user_B_geth.log.zip](https://github.com/status-im/status-mobile/files/13572095/user_B_geth.log.zip)
Note: As we can create communities only on e2e mobile builds I need also a desktop build with this status-go version to check this functionality

**Comment by siddarthkay** [comment]
Thanks for testing this PR @yevh-berdnyk and for summarising the main issues.
I'll create 2 separate issues for these problems and link them here so that I can maybe share the workload with the crew.

**Comment by siddarthkay** [comment]
@yevh-berdnyk : both the issues seem to be fixed with my last commit I believe.
I tried restoring community on x86_64 emulator and could restore the communities you mentioned in test case.
Screenshot 2023-12-07 at 9 24 19 PM

I also followed steps of inviting 2 users as shown by @VolodLytvynenko in the issue description here : https://github.com/status-im/status-mobile/issues/17678 
and after the device that created the community came online, within 1 second the other device's request to join was accepted.
Screenshot 2023-12-07 at 9 15 57 PM

**Comment by yevh-berdnyk** [comment]
> @yevh-berdnyk : both the issues seem to be fixed with my last commit I believe. I tried restoring community on x86_64 emulator and could restore the communities you mentioned in test case. 
> 
> I also followed steps of inviting 2 users as shown by @VolodLytvynenko in the issue description here : #17678 and after the device that created the community came online, within 1 second the other device's request to join was accepted. 

Cool, thank you!
I'll take a look on the other failed tests and will let you know the result soon

**Comment by yevh-berdnyk** [comment]
Hi @siddarthkay,
Thanks a lot for such a big amount of work done here! I've rerun the tests for a few times and now we're good to merge this PR and move to Android14 in e2e tests: #18130

File: ci/Jenkinsfile.android
Comment (siddarthkay) [inline ci/Jenkinsfile.android]: @yakimant : should we get this merged in jenkins repo first?

Comment (yakimant) [reply ci/Jenkinsfile.android]: First, congrats with PR!

Regarding the jenkins lib - it switches the build from 32 to 64 bit.
Which is required to run on Android 12+ emulators.

As I understand you PR fixes issues on 64 bit emulators.
But is it compatible wit 32 bit still?

Another question is switching e2e emulators to Android 12+ 64bit.
Does it happen on SauceLab UI or in our repo?
@yevh-berdnyk, can you help?

Comment (yevh-berdnyk) [reply ci/Jenkinsfile.android]: Hi,
No, it's not compatible with 32 anymore.
Regarding SayceLabs emulators - we're just changing Android version if we need it and that's it. But SauceLabs emulators are x86_64. E2e tests will still run after merging this PR

Comment (siddarthkay) [reply ci/Jenkinsfile.android]: hmm @yevh-berdnyk : thats because the CI for this PR points to this branch of `status-jenkins-lib`
I'm not sure that will be the case if we revert this change.

Also we can't or shouldn't merge this PR without this change first being merged in  `status-jenkins-lib`
probably @jakubgs or @yakimant , @apentori  can confirm...

**Pull request merged**
